src/messages/proposals.rs

```diff
@@ -133,6 +133,12 @@ impl Proposal {
             _ => None,
         }
     }
+    pub(crate) fn as_presharedkey(&self) -> Option<PreSharedKeyProposal> {
```

Comment:
Shouldn't we either take the proposals out of the queue here or (as we do for the others) work with a reference? (That's at least the only call for this I can see.)

Reply:
I also agree with that. The problem is that it would require a bigger refactoring of how the proposal queue works, otherwise we immediately run into lifetime issues. I filed #347.